File: client/rufino/lib/modules/employee/domain/model/role_info/position.dart
import 'package:rufino/modules/employee/domain/model/base/enumeration.dart';
import 'package:rufino/modules/employee/domain/model/base/text_base.dart';
import 'package:rufino/modules/employee/domain/model/role_info/position_cbo.dart';
import 'package:rufino/modules/employee/domain/model/role_info/position_description.dart';

class Position extends Enumeration {
  final PositionDescription positionDescription;
  final PositionCbo positionCbo;

  const Position(
      String id, String name, this.positionDescription, this.positionCbo)
      : super(id, name, "Cargo");
  const Position.empty(
      {this.positionDescription = const PositionDescription.empty(),
      this.positionCbo = const PositionCbo.empty()})
      : super.empty(displayName: "Cargo");

  static Position fromJson(Map<String, dynamic> json) {
    return Position(json["id"], json["name"],
        PositionDescription(json["description"]), PositionCbo(json["cbo"]));
  }

  static List<Position> fromListJson(List<dynamic> listJson) {
    return listJson.map((el) => fromJson(el)).toList();
  }

  List<TextBase> get textBase => [positionDescription, positionCbo];
}

File: server/Services/PeopleManagement/PeopleManagement.Domain/AggregatesModel/PositionAggregate/Position.cs
namespace PeopleManagement.Domain.AggregatesModel.PositionAggregate
{
    public class Position : Entity
    {
        public Name Name { get; private set; } = null!;
        public Description Description { get; private set; } = null!;
        public CBO CBO { get; private set; } = null!;
        public Guid DepartmentId { get; private set; }
        public Guid CompanyId { get; private set; }

        public Position() { }

        public Position(Guid id, Name name, Description description, CBO cBO, Guid departmentId, Guid companyId) : base(id)
        {
            Name = name;
            Description = description;
            CBO = cBO;
            DepartmentId = departmentId;
            CompanyId = companyId;
        }

        public static Position Create(Guid id, Name name, Description description, CBO cBO, Guid departmentId, Guid companyId) => new(id, name, description, cBO, departmentId, companyId);

        public void Edit(Name name, Description description, CBO cBO)
        {
            Name = name;
            Description = description;
            CBO = cBO;
        }
    }
}
